src/app/src/main/java/com/example/bookmark/util/RequestUtil.java

```diff
@@ -0,0 +1,41 @@
+package com.example.bookmark.util;
+
+import android.content.Context;
+
+import com.example.bookmark.models.Book;
+import com.example.bookmark.models.Request;
+import com.example.bookmark.server.StorageServiceProvider;
+
+import java.util.concurrent.Semaphore;
+
+/**
+ * This class implements utility functions related to managing requests
+ *
+ * @author Nayan Prakash
+ */
+public class RequestUtil {
+    public static Request retrieveRequestsOnBookByStatus(Book book, Request.Status status, Context context) {
```

Comment:
I would suggest passing in an `OnSuccessListener<Request>` and an `OnFailureListener` as parameters to make this method asynchronous. I'm a little concern with the current implementation, because as far as I can tell, the app will freeze for the duration of the network call. Users with slow internet may notice their app freezing frequently.